### ISSUE TITLE
[Request] Kernel source for Global Xiaomi 11T (Agate) MTK

### ISSUE THREAD

**Issue by BayuMustikaIP** [MiCode/Xiaomi_Kernel_OpenSource#2419]
Kernel Source Not Release Since 6 month this device launch, please release it

**Comment by kidukarai** [comment]
Please Xiaomi Mi 11T (agate) is an amazing device with a very impressive MTK processor please release the Mi11T (agate) source code :) :+1:

**Comment by feyraftsl** [comment]
@MiCode

**Comment by SAddik13** [comment]
they are not responding

**Comment by cata332** [comment]
@MiCode
@ImRyujinnn Are you manage to build https://github.com/ImRyujinnn/recovery_device_xiaomi_agate ?
**Please relase the recovery !**

**Comment by feyraftsl** [comment]
> @MiCode @ImRyujinnn Are you manage to build https://github.com/ImRyujinnn/recovery_device_xiaomi_agate ? **Please relase the recovery !**

it cannot booting into system, but it booted to twrp

**Comment by cata332** [comment]
@ImRyujinnn Please let me test your recovery...maibe with a custom rom?
Thank you

**Comment by feyraftsl** [comment]
> @ImRyujinnn Please let me test your recovery...maibe with a custom rom? Thank you

https://github.com/ImRyujinnn/Action-Recovery-Builder/releases/download/2620231118/boot.img , if you reboot into system it will booted into twrp again

**Comment by cata332** [comment]
@ImRyujinnn  Android 11 or 12?

**Comment by feyraftsl** [comment]
@cata332 android 12  miui 13.0.1.0 ID based

**Comment by bachtiyar93** [comment]
Why not have response after thread 11 months created?

**Comment by caezaar** [comment]
please xiaomi respond !!

**Comment by Stephanizgo** [comment]
It's well over 15 months still no kernel

**Comment by mansoor4455** [comment]
The biggest mistake in my life is that I chose the Xiaomi device.

**Comment by jyuuren** [comment]
So many people using this phone and still no kernel. Dear Xiaomi PLEASE release kernel source for this device!

**Comment by popy2k14** [comment]
+1 @mi-code

**Comment by bachtiyar93** [comment]
If no have commited by @mi-code  this is last my phone using xiaomi brand.

**Comment by popy2k14** [comment]
Mi too 😬

**Comment by KuroShion** [comment]
> If no have commited by @mi-code this is last my phone using xiaomi brand.

Yeah ,the reason we usually buy xiaomi phone is because of easy access of twrp and rom flashing and other kind of things to play with it ,but since u arent giving the kernel source ,how the heck we are able to do it

**Comment by Stephanizgo** [comment]
Mediatek is brand friendly but any consumer meaning if you buy this phone with mediatek soc, you don't own it because Xiaomi tried to save money on sources. This phone is not very GSI friendly device, GSIs don't boot sometimes and most of the times power button doesn't work on GSIs. We need roms to make this device work and feel fresh again. I'm bored with MIUI, especially the global version. Xiaomi.eu is okay but this device didn't sell in china, which means that we can't get xiaomi.eu on this device (at least not for free), or other Xiaomi.eu based roms like MIUI mind edition ect. I'll hopefully get a new device this year, but I still got hope that they'll release the sources once last OTA arrives or something on this device (could take 2 more years) but I don't recommend getting Mediatek devices at the moment

**Comment by mihancad** [comment]
Hello @MiCode,
According to the GPLv2 license requirements, the kernel source for Xiaomi 11T (agate / amber) must be released publicly.

Without the official kernel source, developers are unable to add EROFS support, build optimized custom kernels, custom ROMs and recoveries (like TWRP/Fastbootd with EROFS/F2FS support).

Please release the full kernel source for Xiaomi 11T on your GitHub under MiCode/kernel repository.

This will help the community improve performance, stability, and extend the life of the device.

Respect Open Source — Respect GPL.

#Xiaomi11T #MiCode #KernelSource #OpenSource #GPLv2 #AndroidKernel #EROFS #Xiaomi
